File: Web.Admin/src/api/services/tenant/models/EditTenantInput.ts
import { CommonStatusEnum } from "@/api/enums/CommonStatusEnum";
import { EditionEnum } from "@/api/enums/EditionEnum";

/**
 * Fast.Center.Service.Tenant.Dto.EditTenantInput 编辑租户输入
 */
export interface EditTenantInput {
  /**
   * 租户Id
   */
  tenantId?: number;
  /**
   * 租户编码
   */
  tenantCode?: string;
  /**
   * 状态
   */
  status?: CommonStatusEnum;
  /**
   * 租户名称
   */
  tenantName?: string;
  /**
   * 租户简称
   */
  shortName?: string;
  /**
   * 租户英文名称
   */
  spellName?: string;
  /**
   * 版本
   */
  edition?: EditionEnum;
  /**
   * 租户管理员名称
   */
  adminName?: string;
  /**
   * 租户管理员手机
   */
  adminMobile?: string;
  /**
   * 租户管理员邮箱
   */
  adminEmail?: string;
  /**
   * 租户管理员电话
   */
  adminPhone?: string;
  /**
   * 租户机器人名称
   */
  robotName?: string;
  /**
   * LogoUrl
   */
  logoUrl?: string;
  /**
   * 行版本
   */
  rowVersion?: number;
}

File: Api.Server/src/Center.Service/Tenant/Dto/EditTenantInput.cs
// Copyright © 2018-Present 小方
// SPDX-License-Identifier: Apache-2.0
// 
// 本文件依据 Apache License 2.0 授权，完整条款见仓库根目录 LICENSE。
// 本软件按“原样”提供，相关免责声明及责任限制以许可证及适用法律为准。
// 版权来源、合法使用与二次开发责任说明见仓库根目录 README.md。

namespace Fast.Center.Service.Tenant.Dto;

/// <summary>
/// 租户编辑输入
/// </summary>
public class EditTenantInput : UpdateVersionInput
{
    /// <summary>
    /// 租户Id
    /// </summary>
    [LongRequired(ErrorMessage = "租户Id不能为空")]
    public long TenantId { get; set; }

    /// <summary>
    /// 租户编码
    /// </summary>
    [StringRequired(ErrorMessage = "租户编码不能为空"), MaxLength(5, ErrorMessage = "租户编码不能超过5个字符")]
    public string TenantCode { get; set; }

    /// <summary>
    /// 状态
    /// </summary>
    [EnumRequired(ErrorMessage = "状态不能为空")]
    public CommonStatusEnum Status { get; set; }

    /// <summary>
    /// 租户名称
    /// </summary>
    [StringRequired(ErrorMessage = "租户名称不能为空")]
    public string TenantName { get; set; }

    /// <summary>
    /// 租户简称
    /// </summary>
    [StringRequired(ErrorMessage = "租户简称不能为空")]
    public string ShortName { get; set; }

    /// <summary>
    /// 租户英文名称
    /// </summary>
    [StringRequired(ErrorMessage = "租户英文名称不能为空")]
    public string SpellName { get; set; }

    /// <summary>
    /// 版本
    /// </summary>
    [EnumRequired(ErrorMessage = "版本不能为空", AllowZero = true)]
    public EditionEnum Edition { get; set; }

    /// <summary>
    /// 租户管理员名称
    /// </summary>
    [StringRequired(ErrorMessage = "租户管理员名称不能为空")]
    public string AdminName { get; set; }

    /// <summary>
    /// 租户管理员手机
    /// </summary>
    [StringRequired(ErrorMessage = "租户管理员手机不能为空")]
    [RegularExpression(RegexConst.Mobile, ErrorMessage = "手机格式不正确")]
    public string AdminMobile { get; set; }

    /// <summary>
    /// 租户管理员邮箱
    /// </summary>
    [StringRequired(ErrorMessage = "租户管理员邮箱不能为空")]
    [RegularExpression(RegexConst.EmailAddress, ErrorMessage = "邮箱格式不正确")]
    public string AdminEmail { get; set; }

    /// <summary>
    /// 租户管理员电话
    /// </summary>
    public string AdminPhone { get; set; }

    /// <summary>
    /// 租户机器人名称
    /// </summary>
    [StringRequired(ErrorMessage = "租户机器人名称不能为空")]
    public string RobotName { get; set; }

    /// <summary>
    /// Logo URL
    /// </summary>
    [StringRequired(ErrorMessage = "LogoUrl不能为空")]
    public string LogoUrl { get; set; }
}
